### PR TITLE
fix: check permissions loading in Page component

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

https://determinedai.atlassian.net/browse/WEB-558

Add `loading` to `usePermissions` hook and check that value in `<Page>` component. Ignore in `<Signin>` page (won't have an authenticated user to fetch permissions).

## Test Plan

In EE/RBAC:
- [ ] Pages should display a spinner until permissions are loaded (instead of flashing inaccurate component state)
- [ ] Logging in/out should still work without getting stuck on a loading spinner

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/usePermissions.ts [hunk 1]
```diff
@@ -75,19 +75,42 @@ interface PermissionsHook {
...
+  const userRoles = Loadable.match(loadableUserRoles, {
+    Loaded: (uRoles) => uRoles,
+    NotLoaded: () => [],
+  });
+
+  const [loading, setLoading] = useState<boolean>(true);
+
+  useEffect(() => {
+    if (
+      !Loadable.isLoading(loadableCurrentUser) &&
```
Comment (ClaireNeveu) [inline webui/react/src/hooks/usePermissions.ts:106]: Does this need to be state?

Comment (johnkim-det) [reply webui/react/src/hooks/usePermissions.ts:106]: i think it does -- tried `useRef` but had issues with it not triggering re-render in `<Page>` component, so spinner doesn't get replaced with content as desired.

Comment (ClaireNeveu) [reply webui/react/src/hooks/usePermissions.ts:106]: Why can't you just return `Loadable.isLoading(loadableCurrentUser) || ...`?

Comment (johnkim-det) [reply webui/react/src/hooks/usePermissions.ts:106]: realized i could do that right after i commented :) updating...

**Comment by ClaireNeveu** [comment]
In your test steps you also need to check loading the page in a logged out state and logging in.

File: webui/react/src/components/Page.tsx [hunk 2]
```diff
@@ -23,6 +27,9 @@ const getFullDocTitle = (branding: string, title?: string, clusterName?: string)
 };
 
 const Page: React.FC<Props> = (props: Props) => {
+  const rbacEnabled = useFeature().isOn('rbac');
```
[the remark this answers is not in the document]
Comment (johnkim-det) [reply webui/react/src/components/Page.tsx]: thanks, will remove that here and include as a condition within `usePermissions`

**Review by mapmeld — APPROVED** [review]
LGTM - ✅  tested in EE, logging in, logging out, reloading trial and workspace pages, etc.

**Pull request merged**